pkg/reports/router.go

```diff
@@ -0,0 +1,75 @@
+package reports
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gorilla/mux"
+	moovhttp "github.com/moov-io/base/http"
+
+	"github.com/moov-io/customers/pkg/client"
+
+	"github.com/moov-io/customers/pkg/accounts"
+	"github.com/moov-io/customers/pkg/customers"
+	"github.com/moov-io/customers/pkg/route"
+)
+
+func AddRoutes(
+	logger log.Logger,
+	r *mux.Router,
+	customerRepo customers.CustomerRepository,
+	accountRepo accounts.Repository,
+) {
+	r.Methods("GET").Path("/reports/accounts").HandlerFunc(getAccounts(logger, customerRepo, accountRepo))
+}
+
+func getAccounts(
+	logger log.Logger,
+	customerRepo customers.CustomerRepository,
+	accountRepo accounts.Repository,
+) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w = route.Responder(logger, w, r)
+		requestID := moovhttp.GetRequestID(r)
+
+		accountIDsInput := r.URL.Query().Get("accountIDs")
+		accountIDsInput = strings.TrimSpace(accountIDsInput)
+		accountIDs := strings.Split(accountIDsInput, ",")
+
+		allAccounts, err := accountRepo.GetCustomerAccountsByIDs(accountIDs)
+		if err != nil {
+			logger.Log("customers", "error getting customers' accounts", "error", err, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+
+		allCustomers := make(map[string]*client.Customer)
+		for _, acc := range allAccounts {
+			cust, err := customerRepo.GetCustomer(acc.CustomerID)
```

Comment:
Need to update `searchCustomers` to accept a list of `customerIDs` to filter for.

New related issue: #220